packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/LightCurveViewer/LightCurveViewerContainer.spec.js

```diff
@@ -1,20 +1,173 @@
+      cdmSpy.resetHistory()
+    })
+
+    after(function () {
+      cdmSpy.restore()
+      nock.cleanAll()
+      nockScope.persist(false)
+    })
+
+    it('should error if a json or text subject location file can\'t be found', function (done) {
+      wrapper = shallow(
+        <LightCurveViewerContainer.wrappedComponent subject={imageSubject} />
+      )
+
+      expect(cdmSpy).to.have.been.calledOnce
+      cdmSpy.returnValues[0].then(() => {
+        expect(wrapper.state().loading).to.equal(asyncStates.error)
+        expect(wrapper.state().error).to.equal('No JSON url found for this subject')
+      }).then(done, done)
+    })
+
+    it('should error if the location request response fails', function (done) {
+      wrapper = shallow(
+        <LightCurveViewerContainer.wrappedComponent subject={failSubject} />
+      )
+
+      cdmSpy.returnValues[0].then(() => {
+        expect(wrapper.state().loading).to.equal(asyncStates.error)
+        expect(wrapper.state().error).to.equal('Not Found')
+      }).then(done, done)
+    })
+
+    it('should set the component state with the json data', function (done) {
+      wrapper = shallow(
+        <LightCurveViewerContainer.wrappedComponent subject={subject} />
+      )
+
+      cdmSpy.returnValues[0].then(() => {
+        wrapper.state().dataExtent.x.forEach((xDataPoint, index) => {
+          expect(xDataPoint).to.equal(mockD3XExtent[index])
+        })
+
+        wrapper.state().dataExtent.y.forEach((yDataPoint, index) => {
+          expect(yDataPoint).to.equal(mockD3YExtent[index])
+        })
+
+        expect(wrapper.state().dataPoints).to.have.lengthOf(mockZipData.length)
+
+        expect(wrapper.state().loading).to.equal(asyncStates.success)
+      }).then(done, done)
+    })
+
+    it('calls the onReady prop on successful load', function (done) {
+      const onReadySpy = sinon.spy()
+      wrapper = shallow(
+        <LightCurveViewerContainer.wrappedComponent onReady={onReadySpy} subject={subject} />
+      )
+      
+      cdmSpy.returnValues[0].then(() => {
+        expect(onReadySpy).to.have.been.calledOnceWith({ target: wrapper.instance().viewer.current })
+      })
+      .then(done, done)
+    })
+
+    it('should update component state when there is a new valid subject prop', function (done) {
+      const nextSubjectJSONZip = zip(nextSubjectJSON.x, nextSubjectJSON.y)
+      const nextSubjectD3XExtent = d3.extent(nextSubjectJSON.x)
+      const nextSubjectD3YExtent = d3.extent(nextSubjectJSON.y)
+      const cduSpy = sinon.spy(LightCurveViewerContainer.wrappedComponent.prototype, 'componentDidUpdate')
+      wrapper = shallow(
+        <LightCurveViewerContainer.wrappedComponent subject={subject} />
+      )
+      wrapper.setProps({ subject: nextSubject })
+
+      // componentDidUpdate gets called three times
+      // only the third is the update to the new props we're expecting for this test
+      // Is there a better way to do this? This seems brittle
```

Comment:
Spying on `wrapper.instance().onLoad` might be what you need here. 

By the way, the classifier app is set up to mount a new subject viewer for each new subject, so these updates may never happen in actual use.

Reply:
Maybe use a timeout here too, and expect the state to have updated after a short pause.